src/components/SendTokens/EthereumTokens.js

```diff
@@ -148,37 +161,43 @@ class SendETHTokens extends React.Component<Props, State> {
     gettingFee: true,
     calculatingMaxValue: false,
     inputHasError: false,
-    txFeeInWei: new BigNumber(0),
+    txFeeInfo: null,
     submitPressed: false,
-    feeByGasToken: false,
     showRelayerMigrationModal: false,
   };
 
   constructor(props: Props) {
     super(props);
 
     this.updateTxFee = debounce(this.updateTxFee, 500);
-    this.updateGasLimitAndTxFee = debounce(this.updateGasLimitAndTxFee, 500);
-    if (!isEmpty(props.gasToken)) this.state.feeByGasToken = true;
+    this.updateKeyWalletGasLimitAndTxFee = debounce(this.updateKeyWalletGasLimitAndTxFee, 500);
   }
 
   componentDidMount() {
-    this.props.fetchGasInfo();
+    if (!this.props.isSmartAccount) {
+      this.props.fetchGasInfo();
+    }
   }
 
   componentDidUpdate(prevProps: Props) {
-    const { session, gasToken, gasInfo } = this.props;
+    const {
+      session, useGasToken, gasInfo, isSmartAccount, isGasTokenSupported,
+    } = this.props;
+    const { showRelayerMigrationModal } = this.state;
 
-    if (prevProps.session.isOnline !== session.isOnline && session.isOnline) {
+    if (prevProps.session.isOnline !== session.isOnline && session.isOnline && isSmartAccount) {
```

Comment:
shouldn't it be `!isSmartAccount`? maybe move this check into `fetchGasInfo` action, so that we don't have to check it every time?

Reply:
Good catch!
I don't think it's worth now to optimise the `fetchGasInfo` action since we're going to remove KW shortly together with that legacy gas fetching